catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,602 @@
+        Metacard metacard = getMetacardUpdatePair().get(0);
+        storeMetacard(metacard);
+
+        // Send a delete request
+        DeleteStorageRequest deleteStorageRequest =
+                new DeleteStorageRequestImpl(Collections.singletonList(metacard), new HashMap<>());
+        storageProvider.delete(deleteStorageRequest);
+
+        // Version delete request
+        DeleteRequest deleteRequest = new DeleteRequestImpl("deleteRequest");
+        DeleteResponse deleteResponse = new DeleteResponseImpl(deleteRequest,
+                new HashMap<>(),
+                Collections.singletonList(metacard));
+        historian.version(deleteResponse);
+
+        // Only the version metacard is left
+        assertThat(storageProvider.storageMap.size(), equalTo(1));
+    }
+
+    @Test
+    public void testDeleteResponseNoContentItems()
+            throws SourceUnavailableException, IngestException, StorageException {
+        Metacard metacard = getMetacardUpdatePair().get(0);
+        DeleteRequest deleteRequest = new DeleteRequestImpl("deleteRequest");
+        DeleteResponse deleteResponse = new DeleteResponseImpl(deleteRequest,
+                new HashMap<>(),
+                Collections.singletonList(metacard));
+
+        historian.version(deleteResponse);
+        assertThat(storageProvider.storageMap.size(), equalTo(0));
+    }
+
+    @Test
+    public void testNullContentItemStorageReadRequest()
+            throws StorageException, UnsupportedQueryException, SourceUnavailableException,
+            IngestException, URISyntaxException {
+        List<Metacard> metacards = getMetacardUpdatePair();
+
+        // Parameters for historian
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        // Make content item null
+        storageProvider.storageMap.put(METACARD_ID, null);
+
+        mockQuery(metacards.get(1));
+        historian.version(storageRequest, storageResponse, updateResponse);
+
+        // Verify that the content item DIDN't updated
+        ReadStorageRequest request = mock(ReadStorageRequest.class);
+        when(request.getResourceUri()).thenReturn(new URI(RESOURCE_URI));
+        ContentItem item = storageProvider.read(request)
+                .getContentItem();
+
+        assertThat(item, nullValue());
+    }
+
+    @Test
+    public void testReadStorageException()
+            throws StorageException, SourceUnavailableException, IngestException,
+            UnsupportedQueryException, URISyntaxException {
+        StorageProvider exceptionStorageProvider = mock(StorageProvider.class);
+        when(exceptionStorageProvider.read(any())).thenThrow(StorageException.class);
+        historian.setStorageProviders(Collections.singletonList(exceptionStorageProvider));
```

Comment:
No, fixed and changed the assertion to make sure that `catalogProvider.create` was never called